src/actions/user-actions.js

```diff
@@ -0,0 +1,84 @@
+import userService from '../services/user-service';
+import {REGISTER_USER, FIND_ALL_USERS, FIND_USER_BY_ID, LOGIN_USER, LOGOUT_USER} from "./user-constants";
+
+export const registerUser = (dispatch, user) => {
+    // retrieve all users
+    userService
+        .findAllUsers()
+        .then(actualUsers => {
+            let flag = 0
```

Comment:
I don't believe a `flag` variable is really needed. Cleaner code would probably be something along the lines of 
```
if user exists: => alert and return or break
else:
createUser with service
```